File: data/readable_format/fortran/file2174.f90
! Define a module for the common blocks and subroutine
module common_blocks
  implicit none
  character(len=15) :: c
  integer :: d, e
  real :: f
end module common_blocks

! Define the subroutine that uses the common blocks
subroutine test()
  use common_blocks
  if ((d .ne. 42) .or. (e .ne. 43) .or. (f .ne. 2.0)) then
    print *, "Test failed due to incorrect values in /block/"
    stop
  endif
  if (c .ne. "Hello World    ") then
    print *, "Test failed due to incorrect value in /block2/"
    stop
  endif
  print *, "All tests passed."
end subroutine test

! Main program to initialize common blocks and call the test
program main
  use common_blocks
  integer :: a(2)
  real :: b
  character(len=15) :: s

  ! Initialize the variables
  a = (/42, 43/)
  b = 2.0
  s = "Hello World    " ! Ensure the string is padded to match the expected length

  ! Assign values from local variables to common block variables
  d = a(1)
  e = a(2)
  f = b
  c = s

  ! Call the test subroutine
  call test()
end program main

File: data/readable_format/cpp/file2174.cpp
#include <iostream>
#include <cstring>
#include <cstdlib>

// Define a structure to represent the common block /block/
struct Block {
    int a[2];
    float b;
} block = {{42, 43}, 2.0};

// Define a structure to represent the common block /block2/
struct Block2 {
    char s[16]; // One extra character to null-terminate the string
} block2 = {"Hello World    "}; // Ensure the string is padded to 15 chars

void abort_program() {
    std::cerr << "Abort called!" << std::endl;
    std::exit(1);
}

void test() {
    // Access common block variables through the structures
    char c[16];
    int d = block.a[0];
    int e = block.a[1];
    float f = block.b;
    std::strcpy(c, block2.s);

    if (d != 42 || e != 43 || f != 2.0f) {
        abort_program();
    }
    if (std::strncmp(c, "Hello World    ", 15) != 0) {
        abort_program();
    }
}

int main() {
    // Call the test function
    test();
    return 0;
}